lib/paper_trail.rb

```diff
@@ -47,32 +48,46 @@ def enabled?
       !!PaperTrail.config.enabled
     end
 
-    # Sets whether PaperTrail is enabled or disabled for the current request.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_controller=(value)
-      paper_trail_store[:request_enabled_for_controller] = value
+      request.enabled_for_controller = value
     end
 
-    # Returns `true` if PaperTrail is enabled for the request, `false` otherwise.
-    #
-    # See `PaperTrail::Rails::Controller#paper_trail_enabled_for_controller`.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_controller?
-      !!paper_trail_store[:request_enabled_for_controller]
+      request.enabled_for_controller?
     end
 
-    # Sets whether PaperTrail is enabled or disabled for this model in the
-    # current request.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_model(model, value)
-      paper_trail_store[:"enabled_for_#{model}"] = value
+      request.enabled_for_model(model, value)
     end
 
-    # Returns `true` if PaperTrail is enabled for this model in the current
-    # request, `false` otherwise.
+    # See PaperTrail::Request for information
     # @api public
     def enabled_for_model?(model)
-      !!paper_trail_store.fetch(:"enabled_for_#{model}", true)
+      request.enabled_for_model?(model)
+    end
+
+    # Returns the PaperTrail::Request object for setting keys for a single request
+    # All methods that handle request-level configuration and are now handled
+    # by PaperTrail::Request
+    #
+    # We will add deprecation notices for these methods in a later version
+    # For now, please refer to PaperTrail::Request for method documentation
+    # @api public
+    def request(options = nil)
+      if options
+        old_request = @request
+        @request ||= PaperTrail::Request.new
+        yield
+        @request = old_request
+      else
+        @request ||= PaperTrail::Request.new
+      end
```

Comment:
I don't think it makes sense use an instance variable here. There's nothing to store. The `Request` instance doesn't have any data of its own, it's just an interface to a `request_store`.